src/libcadet/api/CAPIv1.cpp

```diff
@@ -917,8 +917,16 @@ namespace v1
 			return cdtDataNotStored;
 		}
 
+		bool keepAxialSingletonDimension = unitRec->keepBulkSingletonDim();
+
 		if (nCoords)
 			*nCoords = unitRec->numAxialCells();
+
+		if (nCoords && *nCoords == 1 && !keepAxialSingletonDimension)
+		{
+			return cdtDataNotStored;
```

Comment:
Technically, I'm not sure if `cdtDataNotStored` is the best choice, since the issue is not that it's not stored; instead it should not even exist.

Reply:
I think thats the exception/error that we want